stacks-signer/Cargo.toml

```diff
@@ -42,10 +42,16 @@ tracing = "0.1.37"
 tracing-subscriber = { version = "0.3.17", features = ["env-filter"] }
 wsts = { workspace = true }
 rand = { workspace = true }
+r2d2 = "0.8.9"
```

Comment:
Please just use `rusqlite`

Reply:
Now that I did a first-read of the PR, I think it's safe to say that you really do not need this at all.

Reply:
That's how I started, but because `rusqlite::Connection` isn't thread safe and the signer's runloop is spawned in threads, I couldn't find any good way of managing connections other than opening and closing for every read/write. I'm 100% open to a better alternative - using `r2d2` was really just me finding a solution that worked

Reply:
You'll need to instantiate the `rusqlite::Connection` within the thread, instead of putting yourself in a position of having to pass it between threads.

Reply:
Like, the way we do this elsewhere in the code is we wrap the `rusqlite::Connection` struct within its own `struct`, and then instantiate _that_ struct within the thread that will use it.  These structs aren't singletons; we instead rely on sqlite for ACID guarantees in the event that multiple threads read and write to the DB via separate connections.

Reply:
Thanks for the pointer! Makes sense, I'm working on it.

Reply:
:pray:

Reply:
ah neat! I was wondering exactly how we got around this in stacks block core code.